pulp_container/app/tasks/sync_stages.py

```diff
@@ -428,29 +469,21 @@ async def create_listed_manifest(self, manifest_data):
             name=self.remote.namespaced_upstream_name, digest=digest
         )
         manifest_url = urljoin(self.remote.url, relative_url)
-        manifest = await sync_to_async(
-            Manifest.objects.prefetch_related("contentartifact_set").filter(digest=digest).first
-        )()
-        if manifest:
-
-            def _get_content_data_blocking():
-                saved_artifact = manifest.contentartifact_set.first().artifact
-                content_data = json.load(saved_artifact.file)
-                saved_artifact.file.close()
-                return saved_artifact, content_data
-
-            saved_artifact, content_data = await sync_to_async(_get_content_data_blocking)()
+
+        if digest and (
```

Comment:
in this case digest will be always present because when creating listed manifest we already have parsed and validated json file for manifest_list, which has digest required field for children manifests

Reply:
Probably this was left as one of the experiments to reduce the code repetition.